lib/foreman_inventory_upload/async/remove_insights_hosts_job.rb

```diff
@@ -0,0 +1,67 @@
+module ForemanInventoryUpload
+  module Async
+    class RemoveInsightsHostsJob < ::Actions::EntryAction
+      include ForemanRhCloud::CertAuth
+
+      def plan(search_term, organization_id)
+        plan_self(search_term: search_term, organization_id: organization_id)
+      end
+
+      def run
+        logger.debug("Attempting to remove hosts by search term: #{search_term}")
+
+        host_uuids = InsightsMissingHost.search_for(search_term).pluck(:insights_id)
+
+        page_number = 1
+        while (current_page = host_uuids.paginate(page: page_number, per_page: page_size)).present?
+          logger.debug("Removing #{(page_number - 1) * page_size} - #{page_number * page_size}/#{current_page.total_entries} hosts: #{current_page.join(',')}")
+          response = delete_page(current_page, organization)
+          # write the response in case we want to track it later
+          output["response_page#{page_number}"] = response.body
+
+          # remove host records that reported success after deletion
+          if response.code >= 200 && response.code < 300
```

Comment:
Would any of the status codes besides 200 affect our query or alter the results returned or make our remove operation fail?

Reply:
4XX and 5XX from the cloud would cause the task to fail, but I think it will continue to run, so if the error is intermittent, at least other pages will be processed.
3XX should do the redirects automatically IIRC.